src/core/server/saved_objects/migrationsv2/model/model.ts

```diff
@@ -459,27 +309,46 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   } else if (stateP.controlState === 'WAIT_FOR_YELLOW_SOURCE') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
     if (Either.isRight(res)) {
-      const source = stateP.sourceIndex;
-      const target = stateP.versionIndex;
       return {
         ...stateP,
-        controlState: 'SET_SOURCE_WRITE_BLOCK',
-        sourceIndex: source,
-        targetIndex: target,
-        targetIndexMappings: disableUnknownTypeMappingFields(
-          stateP.targetIndexMappings,
-          stateP.sourceIndexMappings
-        ),
-        versionIndexReadyActions: Option.some<AliasAction[]>([
-          { remove: { index: source.value, alias: stateP.currentAlias, must_exist: true } },
-          { add: { index: target, alias: stateP.currentAlias } },
-          { add: { index: target, alias: stateP.versionAlias } },
-          { remove_index: { index: stateP.tempIndex } },
-        ]),
+        controlState: 'CHECK_UNKNOWN_DOCUMENTS',
       };
     } else {
       return throwBadResponse(stateP, res);
     }
+  } else if (stateP.controlState === 'CHECK_UNKNOWN_DOCUMENTS') {
+    const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+    if (Either.isRight(res)) {
```

Comment:
I'm torn here between returning a `Either.Right` or `Either.Left` for returning something that will ultimately cause the migrations to fail. Generally, the 'happy path' for migrations is indicated by an `Either.Right` res _but_, in this case, out success case is when we have found documents with unknown types.

I guess it's not critical to the implementation, it just feels inconsistent with what I'd have expected.

Reply:
Yea, you're right, using `left` when unknown doc types are found makes sense. Will adapt.